components/kyma-environment-broker/internal/storage/driver/postsql/runtime_state.go

```diff
@@ -171,3 +232,38 @@ func (s *runtimeState) toRuntimeStates(states []dbmodel.RuntimeStateDTO) ([]inte
 
 	return result, nil
 }
+
+func (s *runtimeState) provideClusterSetup(clusterSetup *reconciler.Cluster) ([]byte, error) {
+	marshalledClusterSetup, err := s.marshalClusterSetup(clusterSetup)
+	if err != nil {
+		return nil, errors.Wrap(err, "while encoding reconciler input")
+	}
+	encryptedClusterSetup, err := s.encryptClusterSetup(marshalledClusterSetup)
+	if err != nil {
+		return nil, errors.Wrap(err, "while encrypting reconciler input")
+	}
+	return encryptedClusterSetup, nil
+}
+
+func (s *runtimeState) marshalClusterSetup(clusterSetup *reconciler.Cluster) ([]byte, error) {
+	var (
+		result []byte
+		err    error
+	)
+	if clusterSetup != nil {
+		result, err = json.Marshal(clusterSetup)
+		if err != nil {
```

Comment:
despite being very unlikely, it might be a good idea to at least log the error. I wouldn't be opposed to even going further and after logging of the error, perform retry of the step responsible for storing the operation in `runtime_status` table. Given in https://github.com/kyma-project/control-plane/pull/825, we are going to read what is stored in `runtime_status` last, it could be dangerous not to have the latest information available and KEB could accidentally as a result revert already applied changes.